src/x_connective.c

```diff
@@ -1519,30 +1519,38 @@ typedef struct vcommon
 {
     t_pd c_pd;
     int c_refcount;
-    t_float c_f;
+    t_atom c_atom;
 } t_vcommon;
 
 typedef struct _value
 {
     t_object x_obj;
     t_symbol *x_sym;
-    t_float *x_floatstar;
+    t_atom *x_atomstar;
 } t_value;
 
-    /* get a pointer to a named floating-point variable.  The variable
+    /* get a pointer to a named float variable.  The variable
     belongs to a "vcommon" object, which is created if necessary. */
 t_float *value_get(t_symbol *s)
```

Comment:
Now this is a tricky one. Simply reading the float union member might return garbage (as you've noted), but it can lead to crashes if the client *writes* to the pointer and the actual atom type is a symbol.

This method isn't used in the Pd core outside of this file and I don't know if any externals use it. If an external wants to read/write a float value, it should use `value_setfloat/value_getfloat`. I guess this method is only useful if the external wants to *create* a value object.

One solution could be to bash the value to a float atom:
```
t_atom *a = value_getatom(s);
if (a->a_type != A_FLOAT)
    SETFLOAT(a, 0);
return &a->a_w.w_float;
```

Reply:
>This method isn't used in the Pd core outside of this file

It's in m_pd.h, so who knows what is using it by now, other than expr which is why I added type checks.

Reply:
> other than expr

expr uses `value_getfloat`/`value_setfloat`. I guess `value_get` is only exported for legacy reasons (if at all)... Anyway, I guess your original typecheck + static float would work as well.

Reply:
We could reintroduce my static safety float:

~~~
t_float *value_get(t_symbol *s)
{
    t_atom *a = value_getatom(s);
    if (a->a_type == A_FLOAT)
        return &a->a_w.w_float;
    static t_float f = 0; // safety fallback
    return &f;
}
~~~

Reply:
Added.